### PR TITLE
provider/ec2: add support for ssh keys.

### PR DESCRIPTION
**Pull request by vcaputo**
Keys are added to the "core" user.

### CODE REVIEW

File: src/providers/ec2/ec2.go [hunk 1]
```diff
@@ -101,3 +111,95 @@ func (p provider) ShouldRetry() bool {
...
+	path := fmt.Sprintf(format, a...)
+	data, status, err := p.getData(metadataUrl + path)
+	if err != nil {
+		return nil, err
+	}
+
+	switch status {
+	case http.StatusOK:
+		scanner := bufio.NewScanner(bytes.NewBuffer(data))
+		data := make([]string, 0)
```
Comment (crawford) [inline src/providers/ec2/ec2.go]:
```go
data := []string{}
```

File: src/providers/ec2/ec2.go [hunk 2]
```diff
@@ -101,3 +111,95 @@ func (p provider) ShouldRetry() bool {
...
+		return fmt.Errorf("error reading keys: %v", err)
+	}
+
+	keyIDs := make(map[string]string)
+	for _, keyname := range keynames {
+		tokens := strings.SplitN(keyname, "=", 2)
+		if len(tokens) != 2 {
+			return fmt.Errorf("malformed public key: %q", keyname)
+		}
+		keyIDs[tokens[1]] = tokens[0]
```
Comment (crawford) [inline src/providers/ec2/ec2.go]: We should log the key names as we encounter them. Maybe INFO.

File: src/providers/ec2/ec2.go (hunk 1 above)
Comment (vcaputo) [inline src/providers/ec2/ec2.go]: hm yeah, residual from the cloudinit code

**Comment by crawford** [comment]
LGTM.

**Pull request merged**